operator/v1/types_ingress.go

```diff
@@ -556,6 +571,44 @@ const (
 	AWSNetworkLoadBalancer AWSLoadBalancerType = "NLB"
 )
 
+// AWSSubnets contains a list of references to AWS subnets by
+// ID or name.
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) && has(self.names) ? size(self.ids + self.names) <= 10 : true`,message="the total number of subnets cannot exceed 10"
```

Comment:
This rule needs to be explained in the godoc somewhere, probably on the `subnets` field